plugins/jspsych-spr-moving-window.js

```diff
@@ -399,26 +318,82 @@ var sprMovingWindow = (function(jspsych) {
         const BASE_Y = delta_y; // The height on which lines begin.
         const BASE_X = determineLineHeight(trial_pars.font_family, trial_pars.font_size);
 
-        for (let line = 0; line < lines.length; line++) {
-            liney = BASE_Y + line * delta_y;
-            fragments = lines[line].split(RE_CAP_WHITE_SPACE);
-            fragments = fragments.filter( word => {return word != "";});
-            let runningx = BASE_X;
-            for (let fragment = 0; fragment < fragments.length; fragment++) {
-                let current_fragment = fragments[fragment];
-                let pos = new Pos(runningx, liney);
-                let current_word = new TextInfo(
-                    current_fragment,
-                    pos,
+        let context = {
+            pos : new Pos(BASE_X, BASE_Y),
+        };
+
+        function advanceWhiteSpace(context, ws) {
+            if (ws.get_content() == "\n") {
+                context.pos.x = BASE_X;
+                context.pos.y = context.pos.y + delta_y;
+            }
+            else {
+                let info = new TextInfo(
+                    ws,
+                    new Pos(context.pos.x, context.pos.y),
                     ctx,
                     trial_pars.font_family,
-                    trial_pars.font_size
+                    trial_pars.font_size,
                 );
-                if (!current_word.isWhiteSpace())
-                    words.push(current_word);
-                runningx += current_word.width();
+                context.pos.x += info.width();
             }
         }
+
+        function advanceWord(context, word) {
```

Comment:
I feel like if you made a separate `advanceNewLine` function, then `advanceWhiteSpace` and `advanceWord` can be refactored into a single function (with an extra parameter)

Reply:
I believe I don't really understand what you mean here.

Reply:
the `else` clause of `advanceWhiteSpace` is almost identical to `advanceWord` (with the exception of `words.push(info)`, so I was thinking this could be shared, and the `if` clause of that `else` could become a separate `advanceNewLine`. But feel free to leave it all as is

Reply:
I think I like to keep it this way, as the way its written now maps nicely to the grammar. So if you keep these function next to the grammar.